js/client-mainmenu.js

```diff
@@ -54,10 +55,21 @@
 			buf += '<p><button class="button mainmenu3" name="joinRoom" value="ladder">Ladder</button></p></div>';
 
 			buf += '<div class="menugroup"><p><button class="button mainmenu4 onlineonly disabled" name="joinRoom" value="battles">Watch a battle</button></p>';
-			buf += '<p><button class="button mainmenu5 onlineonly disabled" name="finduser">Find a user</button></p></div>';
+			buf += '<p><button class="button mainmenu5 onlineonly disabled" name="finduser">Find a user</button></p>';
+			buf += "<p class=link-generatore><button id='challeng-via-link' class='button mainmenu5 onlineonly'>send challeng via link</button></p></div>";
 
 			this.$('.mainmenu').html(buf);
-
+			//creating link generatore button with click event to triger link generation event 
+			$('#challeng-via-link').click(function(){if($('.username').length){
+			$('#errormasseg').empty();
+			if(!$('.challenge').length){
+				generateChallengeFourm({opponent_name:"xdxdblue"})
```

Comment:
```suggestion
				generateChallengeForm({opponent_name:"xdxdblue"})
```
Example spelling fix.